src/main/java/net/fabricmc/mappingio/format/StandardProperties.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2023 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.mappingio.format;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.Nullable;
+
+public final class StandardProperties {
+	private StandardProperties() {
+	}
+
+	public static Set<StandardProperty> values() {
+		return Collections.unmodifiableSet(values);
+	}
+
+	@Nullable
+	public static StandardProperty getByName(String name) {
+		return valuesByName.get(name);
+	}
+
+	@Nullable
+	@ApiStatus.Internal
+	public static StandardProperty getById(String id) {
+		return valuesById.get(id);
+	}
+
+	public static final StandardProperty NEXT_INTERMEDIARY_CLASS;
+	public static final StandardProperty NEXT_INTERMEDIARY_FIELD;
+	public static final StandardProperty NEXT_INTERMEDIARY_METHOD;
+	public static final StandardProperty NEXT_INTERMEDIARY_COMPONENT;
```

Comment:
Im not sure these should exist in mio, they are specific to intermediary and are generated by matcher. They arent part of the tiny format, and I dont think we actually read them anymore?

Reply:
happy to discuss this if you think otherwise.

Reply:
I'd say this is a bit of an edge-case, since the counters [have](https://github.com/FabricMC/mapping-io/blob/master/src/main/java/net/fabricmc/mappingio/format/tiny/Tiny1FileReader.java?rgh-link-date=2023-09-24T12%3A43%3A44Z#L183-L185) [been](https://github.com/FabricMC/stitch/blob/0abaedb727630b679224667f3cc8950428db8f7d/src/main/java/net/fabricmc/stitch/commands/GenState.java#L489) special-cased until now and consequently behaved as de-facto standard properties. I could remove it, so we end up using the `INTERMEDIARY_COUNTER` notation in Tiny v2, too, though this would conflict with existing naming conventions and we would be stuck with a misleading name. We could of course also delegate this responsibility to Matcher, which would have to do the conversion via a custom forwarding visitor pass

Reply:
In the long run we might need an API letting consumers register their own standard properties